Больше деталей

urchifox opened the pull request:
---
:mortar_board: [Больше деталей](https://up.htmlacademy.ru/javascript/29/user/2317955/tasks/9)

urchifox committed:
WIP 4.16. Больше деталей

urchifox committed:
make module4-task1

urchifox committed:
4.16. Больше деталей - make another version

urchifox committed:
4.16. Больше деталей refactor and correct mistakes

sanjoyz commented on js/main.js:
```diff
@@ -0,0 +1,179 @@
+  const ids = getNumbersArray(1, fotosAmount, getRandomIntegerNotNegativeNumber);
+  const urls = getNumbersArray(1, fotosAmount, getRandomIntegerNotNegativeNumber);
+  const fotos = [];
+  for (let i = 0; i < fotosAmount; i++) {
```
return ids.map((id, index) => ({
      id: id,
      url: `photos/${urls[index]}.jpg`,
      description: getRandomElement(DESCRIPTIONS_SET),
      likes: getRandomIntegerNotNegativeNumber(15, 200),
      comments: getRandomComments(30),
})

urchifox committed:
4.16. Больше деталей refactoring

Сначала я переделала функцию getRandomFotos, чтобы массив генерился через .map(). Но затем я упростила эту функцию (а заодно и getRandomComments), сгенерив id через обычный счетчик (тз это позволяет). Благодаря этому стало возможным отказаться от вспомогательной функции  getNumbersArray.

sanjoyz commented on js/main.js:
```diff
@@ -79,61 +79,38 @@ const getMessage = (minSize, maxSize) => {
-  for (let i = 0; i < commentsAmount; i++) {
-    const comment = {
-      id: ids[i],
+  return Array.from({length: commentsAmount}, (element, id) => {
```
id это по идее index (можно i), т.е. порядкойвый номер элемента в массиве.

[параметры коллбека для map](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/map#%D0%BF%D0%B0%D1%80%D0%B0%D0%BC%D0%B5%D1%82%D1%80%D1%8B)

sanjoyz commented on js/main.js:
```diff
@@ -79,61 +79,38 @@ const getMessage = (minSize, maxSize) => {
-    const comment = {
-      id: ids[i],
+  return Array.from({length: commentsAmount}, (element, id) => {
+    id++;
```
я бы вынес это непосредственно в 
```
return {
id: ++id
}
```

sanjoyz replied:
или id: id + 1, что понятнее

urchifox committed:
4.16. Больше деталей correct parameters in getRandomComments and getRandomFotos
